### DISCUSSION TITLE
Streetnames on map

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#214] [Q&A] [answered]
It's really not a problem - but I'm curious ...
I saw in low zoom levels (10 or 9) that some "minor" roads on the map are named - other not, although they are highways. Here two examples:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/7cd3140b-9431-40d6-a90a-ade5cd654b56"/>   !<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/c92c2316-f29c-4ffc-bf43-f2a547554484"/> 
The M9 in particular seems to be a favorite. 😉 Why is that? Is it due to definitions in the OSM? 🤔
Thanks for an explanation.

**Comment by devemux86** [comment]
@WolfgangZ42 interesting question!

(besides the road types via OpenStreetMap tags and their priority in map rendering)

An important thing is the **available space** to render the labels, with collision detection techniques, etc.

In this example, maybe the M9 (with its very small name) finds the available space to display its label?

This is one area in map engines where there is always room for improvement. :slightly_smiling_face: 

The same also applies for the labels of cities.
There are cases in all map engines where a small city shows its label earlier than a more important city.

**Reply by MotoUKRider** [reply]
As this looks like my theme, I'll add that I don't recall modifying any Priority values so you *should* see the same behaviour with the "Biker" theme.  If there's something (or even lots) that is really "silly" and needs a priority shift (up or down), perhaps I can view it here and try some numbers?

**Reply by devemux86** [reply]
If anyone wants to play with the priorities in the render themes, please see the relevant section in the render theme documentation:

- https://github.com/mapsforge/vtm/blob/master/docs/Rendertheme.md#priorities

---

@MotoUKRider and everyone,

Please remember to make changes to the latest version of the map themes from the repository:

- https://github.com/mapsforge/vtm/tree/master/vtm-themes/resources/assets/vtm

Then it's much easier for me to update the themes in Cruiser! :slightly_smiling_face:

**Reply by WolfgangZ42** [reply, marked as answer]
> the same behaviour with the "Biker" theme

That's true! As I said - I was just curious. My question is answered - thanks. 👍🙋‍♂️
